src/main/java/org/opentripplanner/transit/service/DefaultTransitService.java

```diff
@@ -551,31 +551,29 @@ private TimetableSnapshot lazyGetTimeTableSnapShot() {
   }
 
   @Override
-  public TripOnServiceDate getTripOnServiceDateById(FeedScopedId datedServiceJourneyId) {
+  public TripOnServiceDate getTripOnServiceDateById(FeedScopedId tripOnServiceDateId) {
     TimetableSnapshot currentSnapshot = lazyGetTimeTableSnapShot();
     if (currentSnapshot != null) {
       TripOnServiceDate tripOnServiceDate = currentSnapshot.getRealTimeAddedTripOnServiceDateById(
-        datedServiceJourneyId
+        tripOnServiceDateId
       );
       if (tripOnServiceDate != null) {
         return tripOnServiceDate;
       }
     }
-    return transitModelIndex.getTripOnServiceDateById().get(datedServiceJourneyId);
+    return transitModel.getTripOnServiceDateById(tripOnServiceDateId);
   }
 
   @Override
   public Collection<TripOnServiceDate> getAllTripOnServiceDates() {
     TimetableSnapshot currentSnapshot = lazyGetTimeTableSnapShot();
     if (currentSnapshot != null) {
       return new CollectionsView<>(
-        transitModelIndex.getTripOnServiceDateForTripAndDay().values(),
+        transitModel.getAllTripOnServiceDates(),
```

Comment:
Shouldn't it be `getAllTrips` instead of `getAllTrip` both here and in the transit model?